lib/NormalModuleReplacementPlugin.js

```diff
@@ -16,7 +16,7 @@ class NormalModuleReplacementPlugin {
 		const resourceRegExp = this.resourceRegExp;
 		const newResource = this.newResource;
 		compiler.hooks.normalModuleFactory.tap("NormalModuleReplacementPlugin", (nmf) => {
-			nmf.plugin("before-resolve", (result, callback) => {
+			nmf.hooks.beforeResolve.tapAsync("NormalModuleReplacementPlugin", (result, callback) => {
```

Comment:
Use `tap` instead of `tapAsync` and remove the `callback`. Return the result instead.